### PR TITLE
Remove virtual functionality from internal tagger api that was never overridden

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs [hunk 1]
```diff
@@ -524,7 +524,7 @@ private static bool StatementClosingDelimiterIsMissing(SyntaxNode currentNode)
         /// </returns>
         private static bool RequiredDelimiterIsMissing(SyntaxNode currentNode)
         {
-            return currentNode.GetBrackets().closeBrace.IsMissing ||
+            return currentNode.GetBrackets().closeBracket.IsMissing ||
                 currentNode.GetParentheses().closeParen.IsMissing;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs:528]: this is just the buildbreak for https://github.com/dotnet/roslyn/pull/52704 merged into thsi branch.

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs [hunk 2]
```diff
@@ -92,7 +90,6 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
             private readonly CancellationTokenSource _initialComputationCancellationTokenSource = new();
 
             public TaggerDelay AddedTagNotificationDelay => _dataSource.AddedTagNotificationDelay;
-            public TaggerDelay RemovedTagNotificationDelay => _dataSource.RemovedTagNotificationDelay;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs:95]: both these members always had the same value provided (and never overridden). So they could be removed, and hteir value inlined into the locations where they were previously read.

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs [hunk 3]
```diff
@@ -112,7 +109,6 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
                 _dataSource = dataSource;
                 _asyncListener = asyncListener;
                 _notificationService = notificationService;
-                _tagSpanComparer = new TagSpanComparer(_dataSource.TagComparer);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs:115]: no need for a separate field/instance here at all

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs [hunk 4]
```diff
@@ -302,80 +298,6 @@ private void RaiseResumed()
 
             private static T NextOrDefault<T>(IEnumerator<T> enumerator)
                 => enumerator.MoveNext() ? enumerator.Current : default;
-
-            /// <summary>
-            /// Return all the spans that appear in only one of "latestSpans" or "previousSpans".
-            /// </summary>
-            private static DiffResult Difference<T>(IEnumerable<ITagSpan<T>> latestSpans, IEnumerable<ITagSpan<T>> previousSpans, IEqualityComparer<T> comparer)
-                where T : ITag
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs:310]: this moved from a static/generic member to just a simple instance non-static member.  Ther was no need for generic as this is always instantiated with the outer TTag type that we already ahve access to.  It also is only called from one exact location, so even having it be shared and available for other callers is pointless.

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_IEqualityComparer.cs [hunk 5]
```diff
@@ -6,30 +6,19 @@
...
 using System.Collections.Generic;
 using Microsoft.VisualStudio.Text.Tagging;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Tagging
 {
     internal abstract partial class AbstractAsynchronousTaggerProvider<TTag>
     {
-        private class TagSpanComparer : IEqualityComparer<ITagSpan<TTag>>
+        private partial class TagSource : IEqualityComparer<ITagSpan<TTag>>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_IEqualityComparer.cs:15]: the TagSource itself just became the IEqualityComparer.

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs [hunk 6]
```diff
@@ -618,12 +618,81 @@ private void ProduceTagsSynchronously(TaggerContext<TTag> context)
...
+            /// </summary>
+            private static DiffResult ComputeDifference(
                 ITextSnapshot snapshot,
-                TagSpanIntervalTree<TTag> latestSpans,
-                TagSpanIntervalTree<TTag> previousSpans)
+                TagSpanIntervalTree<TTag> latestTree,
+                TagSpanIntervalTree<TTag> previousTree)
             {
-                return Difference(latestSpans.GetSpans(snapshot), previousSpans.GetSpans(snapshot), _dataSource.TagComparer);
+                var latestSpans = latestTree.GetSpans(snapshot);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs:629]: this logic was just inlined from the other file.  No other changes were made.

**Review by davidwengier — APPROVED** [review]

**Pull request merged**